druid/src/tests/mod.rs

```diff
@@ -14,18 +14,30 @@
 
 //! Additional unit tests that cross file or module boundaries.
 
-mod harness;
+pub(crate) mod harness;
 mod helpers;
 mod layout_tests;
 
 use std::cell::Cell;
+use std::fs;
 use std::rc::Rc;
 
 use crate::widget::*;
 use crate::*;
 use harness::*;
 use helpers::*;
 
+// These functions are currently only used behind features but may be used else were in the future
+#[allow(dead_code)]
+const TEST_IMAGE_OUTPUT_DIR: &str = "target/test_images";
```

Comment:
maybe the CARGO_TARGET_DIR environment variable?

I'm unsure if that isn't explicitly set by the user if it would contain anything here though.
but e.g. here where I set it to "~/tmp/target/" doesn't generally exist relative to manifest

Reply:
I believe that is read by cargo, but not set by cargo? If there's a way to get it it would be perfect.

Reply:
Dose it matter were in target this resolves too? I think this should always end up in target for testing https://github.com/rust-lang/rustup/blob/master/tests/mock/clitools.rs#L82 and we could even make a temp dir to minimise the chance of old tests giving false positives.

Reply:
I think the current version of the code dose what you want.

Reply:
lgtm